Bottom: 80b5b887e1ce3ef4f40cc7426c93afc08b0c3918
Top:    6290ca3841bc05f674ae3e335bb0a102abe13654
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-16 13:45:22 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 82baaf0..aa29d3d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1120,7 +1120,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
